diff --git meld/meld/build_helpers.py meld/meld/build_helpers.py
index 3240d667c2a1cbc326edcac7252a677ccc7b38dd..0d67f9b3e4a0a321c8af6ce3d76080b17e1237ec 100644
--- meld/meld/build_helpers.py
+++ meld/meld/build_helpers.py
@@ -41,6 +41,11 @@ except ImportError:
             'install paths may be incorrect', file=sys.stderr)
 
 windows_build = os.name == 'nt'
+if windows_build:
+    import cx_Freeze
+    command_base = cx_Freeze
+else:
+    command_base = distutils
 
 
 def has_help(self):
@@ -59,7 +64,7 @@ def has_data(self):
     return "build_data" in self.distribution.cmdclass
 
 
-distutils.command.build.build.sub_commands.extend([
+command_base.command.build.Build.sub_commands.extend([
     ("build_i18n", has_i18n),
     ("build_icons", has_icons),
     ("build_help", has_help),
